OptimizelySDK/OptimizelyDecisionContext.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * Copyright 2021, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OptimizelySDK
+{
+    /// <summary>
+    /// OptimizelyDecisionContext contains flag key and rule key to be used for setting 
+    /// and getting forced decision.
+    /// </summary>
+    public class OptimizelyDecisionContext
+    {
+        public const string OPTI_NULL_RULE_KEY = "$opt-null-rule-key";
+        public const string OPTI_KEY_DIVIDER = "-$opt$-";
+
+        private string flagKey;
+        private string ruleKey;
+        private string decisionKey;
+
+        public string FlagKey { get { return flagKey; } }
+        public string RuleKey { get { return ruleKey; } }
+
+        public OptimizelyDecisionContext(string flagKey, string ruleKey= null)
```

Comment:
Can we add a check here to confirm flagKey is set? rather than in the user context? In Java we have @NotNull we can add before parameters that must be populated. In csharp can we do something alone those lines or add a is null check here ? Then we only need to confirm the context is valid in the OptimizelyUserContext class when interacting with forced decisions. I think this would be a bit cleaner.

Reply:
Done.